ctru-rs/examples/linear-memory.rs

```diff
@@ -0,0 +1,46 @@
+#![feature(allocator_api)]
+
+use ctru::linear::LinearAllocator;
+use ctru::prelude::*;
+
+fn main() {
+    ctru::init();
+    let gfx = Gfx::init().expect("Couldn't obtain GFX controller");
+    let hid = Hid::init().expect("Couldn't obtain HID controller");
+    let apt = Apt::init().expect("Couldn't obtain APT controller");
+    let _console = Console::init(gfx.top_screen.borrow_mut());
+
+    let linear_space_before = LinearAllocator::free_space();
+
+    // Normal `Box` on the heap
+    let heap_box = Box::new(1492);
+    // `Box` living on the linear memory sector
+    let linear_box: Box<i32, LinearAllocator> = Box::new_in(2022, LinearAllocator);
```

Comment:
Is adding this explicit type needed?

Reply:
Not really, but I believe it’s better to have it explicit, since the `Box` type is usually believed to be just `Box<T>`. It’s just an example, Rust can infer the type obviously.

Reply:
Alright, makes sense